libs/ledger-live-common/src/families/cosmos/api/Cosmos.unit.test.ts

```diff
@@ -0,0 +1,797 @@
+      mockedNetwork.mockResolvedValue({
+        data: {
+          account: {
+            "@type": "/cosmos.auth.v1beta1.BaseAccount",
+            account_number: 1,
+            sequence: 0,
+            pub_key: { key: "k", "@type": "type" },
+          },
+        },
+      } as AxiosResponse);
+
+      const account = await cosmosApi.getAccount("addr");
+      expect(account).toEqual({
+        accountNumber: 1,
+        sequence: 0,
+        pubKey: "k",
+        pubKeyType: "type",
+      });
+
+      mockedNetwork.mockResolvedValue({
+        data: {
+          account: {
+            "@type": "/ethermint.types.v1.EthAccount",
+            account_number: 1,
+            sequence: 0,
+            pub_key: { key: "k", "@type": "type" },
+
+            base_account: {
+              address: "address",
+              pub_key: { key: "k2", "@type": "type2" },
+              account_number: 2,
+              sequence: 3,
+            },
+            code_hash: "codeHash",
+          },
+        },
+      } as AxiosResponse);
+
+      const ethermintAccount = await cosmosApi.getAccount("addr");
+      expect(ethermintAccount).toEqual({
+        accountNumber: 2,
+        sequence: 3,
+        pubKey: "k2",
+        pubKeyType: "type2",
+      });
+    });
+
+    it("should return default sequence value if network fails", async () => {
+      mockedNetwork.mockImplementation(() => {
+        throw new Error();
+      });
+      const account = await cosmosApi.getAccount("addr");
+      expect(account.sequence).toEqual(0);
+    });
+
+    it("should return default pubkeytype value if network fails", async () => {
+      mockedNetwork.mockImplementation(() => {
+        throw new Error();
+      });
+      const account = await cosmosApi.getAccount("addr");
+      expect(account.pubKeyType).toEqual(cryptoFactory("cosmos").defaultPubKeyType);
+    });
+  });
+
+  describe("simulate", () => {
+    it("should return gas used when the network call returns gas used", async () => {
+      mockedNetwork.mockResolvedValue({
+        data: {
+          gas_info: {
+            gas_used: 42000,
+          },
+        },
+      } as AxiosResponse);
+      const gas = await cosmosApi.simulate([]);
+      expect(gas).toEqual(new BigNumber(42000));
+    });
+
+    it("should throw an error when the network call does not return gas used", async () => {
+      mockedNetwork.mockResolvedValue({
+        data: { gas_info: {} },
+      } as AxiosResponse);
+      await expect(cosmosApi.simulate([])).rejects.toThrowError();
+    });
+
+    it("should throw an error when the network call fails", async () => {
+      mockedNetwork.mockImplementation(() => {
+        throw new Error();
+      });
+      await expect(cosmosApi.simulate([])).rejects.toThrowError();
+    });
+  });
+
+  describe("getTransactions", () => {
+    it("should return an empty array when network call fails", async () => {
+      // @ts-expect-error method is mocked
```

Comment:
I think we can drop this by using mockedNetwork and it just hasn't been done everywhere... if it doesn't work let's not change it as it's not really related to the PR

Reply:
👍